src/main/scala/mesosphere/marathon/api/v2/AppsResource.scala

```diff
@@ -349,7 +372,13 @@ class AppsResource @Inject() (
       case (_, false) => CompleteReplacement
     }
 
+    val roleSettings = RoleUtils.getRoleSettingsForService(config, appId, groupManager.rootGroup())
+    val normalizationConfig = AppNormalization.Configuration(config, roleSettings)
+    implicit val appDefinitionValidator: Validator[AppDefinition] = AppDefinition.validAppDefinition(config.availableFeatures, roleSettings)(pluginManager)
+    implicit val validateAndNormalizeApp: Normalization[raml.App] = appNormalization(normalizationConfig)(AppNormalization.withCanonizedIds())
```

Comment:
There's lots or repetition. What do you think about moving this into a helper method?

Reply:
Yeah, moved the validator/normalizer-creation into two helper methods, one for apps one for appUpdates